rs/nns/governance/src/governance.rs

```diff
@@ -4046,7 +4049,19 @@ impl Governance {
         //
         // See "Recommendations for Using `unsafe` in the Governance canister" in canister.rs
         let governance: &'static mut Governance = unsafe { std::mem::transmute(self) };
-        spawn(governance.perform_action(pid, action.clone()));
+
+        #[cfg(target_arch = "wasm32")]
+        {
+            spawn(governance.perform_action(pid, action.clone()));
+        }
+        // This is needed for tests
+        #[cfg(not(target_arch = "wasm32"))]
+        {
+            governance
+                .perform_action(pid, action.clone())
+                .now_or_never()
+                .expect("Future could not execute in non-WASM environment");
+        }
```

Comment:
I'm not sure this is the right place to put this logic. Maybe we could have a `spawn` function as as part of the `Environment` trait, and then implement it with `spawn()` for NativeEnvironment and with `now_or_never` for the MockEnvironment. (Names are from memory so might be incorrect but hopefully you get the idea.) 

NCR

Reply:
I'll make a note to follow up on this.